src/main.cpp

```diff
@@ -188,13 +189,16 @@ int main(int argc, char* argv[])
     spDebugConsole = nullptr;
     unsigned int startupAction = 0;
 
-#ifdef Q_OS_UNIX
-    QApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
-#endif
-
     QScopedPointer<QCoreApplication> initApp(createApplication(argc, argv, startupAction));
     auto * app = qobject_cast<QApplication*>(initApp.data());
 
+#ifdef Q_OS_UNIX
+    auto dpi = app->primaryScreen()->devicePixelRatio();
+    if (dpi > 1) {
```

Comment:
`dpi` is a `qreal` so the comparison should be something like `if (dpi > 1.01) {` shouldn't it?

Also - **this must be move back up to before the call to `initApp(...)`** as that is where the `QGuiApplication` singleton instance is(or rather *may*) be created and the documentation says it has to set before then.

Finally - do we really need to code this in? The user can get the same effect with that `QT_AUTO_SCREEN_SCALE_FACTOR` environment variable being set to `1`.

Reply:
The user should not have to fiddle with any environment variables for Mudlet to work right!